pvlib/tests/test_spectrum.py

```diff
@@ -317,6 +318,62 @@ def test_spectral_factor_caballero_supplied_ambiguous():
                                            coefficients=None)
 
 
+@pytest.mark.parametrize("module_type,expected", [
+    ('asi', np.array([1.15534029, 1.1123772, 1.08286684, 1.01915462])),
+    ('fs-2', np.array([1.0694323, 1.04948777, 1.03556288, 0.9881471])),
+    ('fs-4', np.array([1.05234725, 1.037771, 1.0275516, 0.98820533])),
+    ('multisi', np.array([1.03310403, 1.02391703, 1.01744833, 0.97947605])),
+    ('monosi', np.array([1.03225083, 1.02335353, 1.01708734, 0.97950110])),
+    ('cigs', np.array([1.01475834, 1.01143927, 1.00909094, 0.97852966])),
+])
+def test_spectral_factor_pvspec(module_type, expected):
+    ams = np.array([1.0, 1.5, 2.0, 1.5])
+    kcs = np.array([0.4, 0.6, 0.8, 1.4])
+    out = spectrum.spectral_factor_pvspec(ams, kcs,
+                                          module_type=module_type)
+    assert np.allclose(expected, out, atol=1e-8)
+
+
+@pytest.mark.parametrize("module_type,expected", [
+    ('asi', pd.Series([1.15534029, 1.1123772, 1.08286684, 1.01915462])),
+    ('fs-2', pd.Series([1.0694323, 1.04948777, 1.03556288, 0.9881471])),
+    ('fs-4', pd.Series([1.05234725, 1.037771, 1.0275516, 0.98820533])),
+    ('multisi', pd.Series([1.03310403, 1.02391703, 1.01744833, 0.97947605])),
+    ('monosi', pd.Series([1.03225083, 1.02335353, 1.01708734, 0.97950110])),
+    ('cigs', pd.Series([1.01475834, 1.01143927, 1.00909094, 0.97852966])),
+])
+def test_spectral_factor_pvspec_series(module_type, expected):
+    ams = pd.Series([1.0, 1.5, 2.0, 1.5])
+    kcs = pd.Series([0.4, 0.6, 0.8, 1.4])
+    out = spectrum.spectral_factor_pvspec(ams, kcs,
+                                          module_type=module_type)
+    if isinstance(out, pd.Series):
+        assert np.allclose(expected, out, atol=1e-8)
```

Comment:
That will only execute the values assertion if the output type is a `pd.Series`. It won't check that the output is actually that object.

```suggestion
    assert isinstance(out, pd.Series)
    assert np.allclose(expected, out, atol=1e-8)
```

Reply:
ohh... my (original) thinking was that if the output is not `pd.Series` then because the values assertion doesn't execute, the test fails. 
I have committed your suggestion. Thanks

Reply:
Don't worry. Just in case, a test only fails when an exception of any kind is raised (and it is not inside an exception block, like a `pytest.raises` to check that an exception raises -> that block raises an exception if no exception is raised).